core/experiments/manager.php

```diff
@@ -594,6 +599,15 @@ private function render_feature_dependency( $feature ) {
 		<?php
 	}
 
+	private function has_non_existing_dependency( $feature ) {
+		$non_existing_dep = ( new Collection( $feature['dependencies'] ?? [] ) )
+			->find( function ( $dependency ) {
+				return $dependency instanceof Non_Existing_Dependency;
+			} );
```

Comment:
do you still need the `Non_Existing_Dependency` instance if you're hiding the experiment entirely? Can't you just return null?

Reply:
It feels weird to me checking if there is a null in the array and this is an indicator that experiment is not exists.